numpy/lib/tests/test_io.py

```diff
@@ -1903,14 +1903,16 @@ def test_load_refcount():
 
     f = BytesIO()
     np.savez(f, [1, 2, 3])
-    f.seek(0)
 
-    gc.collect()
-    n_before = len(gc.get_objects())
-    np.load(f)
-    n_after = len(gc.get_objects())
-
-    assert_equal(n_before, n_after)
+    # Try a few times to make sure we don't miss stuff accidentally (e.g. an
+    # automatic gc could happen before we can run our manual one)
+    for i in range(10):
+        f.seek(0)
+        gc.collect()
+        np.load(f)
+        # gc.collect returns the number of unreachable objects in cycles that
+        # were found -- we are checking that no cycles were created by np.load
```

Comment:
may be we can disable/enable gc to prevent automatic call, something like:
```python
    gc.disable()
    try:
        gc.collect()
        np.load(f)
        nobj=gc.collect()
    finally:
        gc.enable()
    assert_equal(nobj, 0)
```